iroh/src/magicsock.rs

```diff
@@ -1965,6 +2048,9 @@ impl Actor {
             }
         }
 
+        // kick off resolving the URLs for relay addresses
+        self.resolve_qad_addrs(Duration::from_millis(10)).await;
```

Comment:
AFAIK net-report does do the address lookup anyway.  Is there any way that it could feed this address back to the magicsock?

Brainstorming, but if you add a new `MagicSock::ip_mapped_addrs: Mutex<BTreeMap<u64, SocketAddr>>` because of the other comment then you could as well put that in an arc: `<Arc<Mutex<BTreeMap<u64, SocketAddr>>>` and then you could pass it into net-report and it could directly call `add_quad_addr` on it (well, you'd need to wrap it into a named type with an impl but that's probably a good idea anyway).

An alternative way of hooking up these two together is to send a message to the `magicsock::Actor` to update the "ip_mapped_addrs".  But that would need to have a one-shot reply to signal completion since this needs to be synchornised.

Reply:
What I like about these options is that there's a direct link between those two rather than a DNS lookup that's not yet needed happening for obscure reasons.